app/controllers/placements/providers/placements_controller.rb

```diff
@@ -22,6 +21,22 @@ def show
 
   private
 
+  def set_schools
+    @schools = partner_schools_present? ? partner_schools : all_placements_schools
+  end
+
+  def partner_schools_present?
+    filter_params[:partner_schools]&.compact_blank.present?
+  end
+
+  def partner_schools
+    @provider.partner_schools.select(:id, :name)
+  end
+
+  def all_placements_schools
+    Placements::School.select(:id, :name)
+  end
```

Comment:
I feel like this could probably be collapsed a bit.

```suggestion
  def schools_scope
    if filter_params[:only_partner_schools]
      @provider.partner_schools
    else
      Placements::School.all
    end
  end
```